flocker/control/httpapi.py

```diff
@@ -492,11 +498,15 @@ def create_container_configuration(self, host, name, image, ports=()):
                 external_port=port['external']
             ))
 
+        if environment:
```

Comment:
I suspect it actually can be `None` though since it's not defined as required by the schema.  If no environment is supplied at all then it should come through as `None`.